dolma/src/lib/decode.ts

```diff
@@ -1,12 +1,13 @@
+import dolma from "../index";
 import { Unitoken } from "../tokens";
 import { MessageToken } from "../util/types/tokenTypes";
 import { encodeTokens } from "./encode";
 
-export function decodeTokens(all: Array<Unitoken | MessageToken | string> | string): string {
-	const tokens = encodeTokens(all);
+export function decodeTokens(this: {emotes: {name: string}[]}, all: Array<Unitoken | MessageToken | string> | string): string {
+	const tokens = new dolma(this.emotes).encode(all);
 	let vals: string[] = [];
 
-	tokens.map(tkn => {
+	tokens.tokens.map(tkn => {
 		if (tkn.t == 'text') return vals.push(tkn.v);
 		if (tkn.t == 'block') return vals.push(`\`${tkn.v}\``);
 		if (tkn.t == 'emote') return vals.push(`:${tkn.v}:`);
```

Comment:
please add `emoji` token to decoding aswell

Reply:
the decode function is broken, it will be fixed in a separate pr